discussions/prog_repo.md

```diff
@@ -0,0 +1,192 @@
+# L3AF eBPF Package Repository
+
+The concept of a L3AF eBPF Package Repository is to create a location where
+eBPF Programs from any trusted party can be uploaded and made available for others to download.
+
+In the context of L3AF, we define an eBPF Package as a kernel space program with
+an optional, cooperative user space program.
+
+# What should we name it
+
+"eBPF Package Repository" is the name that has been chosen by the Technical Steering Committee.
+
+# Is the eBPF Package Repository part of the L3AF Project
+
+Firstly, we define the "L3AF Project" as the L3AF open-source project that exists
+within The Linux Foundation. When we say "L3AF", we are referring to the entire L3AF Project.
+This is not to be confused with:
+
+- The `l3af-project` GitHub organization
+- The L3AFD daemon, which is a major component of the L3AF Project.
+
+We consider L3AF to be an entire ecosystem that aims to provide eBPF Programs as a service.
+We've otherwise phrased this as "complete lifecycle management of eBPF programs".
+This definition includes:
+
+- The L3AFD daemon, an orchestrator that provides APIs to launch and manage eBPF programs on a node
+- The L3AF eBPF Package Repository
+- Programs within the eBPF Package Repository
+
+Note, however, that we have no intention of limiting the creation of other public or private
+eBPF Package repositories.
+
+We recognize that it may make sense to migrate the eBPF Package Repository
+(eBPF Programs within the repository) out of the L3AF Project and into its own
+Linux Foundation project in the future. Doing this initially may not make sense (due to L3AF-specific
+eBPF program chaining implementation for networking programs), but as L3AF and other projects mature,
+a platform-agnostic repository could be useful for multiple projects.
+
+# What should an initial version look like
+
+This section examines simple ways to create a location where eBPF Programs can be uploaded
+and made available for others to download.
+
+## A GitHub repository may be sufficient
+
+We would like to leverage GitHub:
+
+- A new GitHub repository will be created to store eBPF program source code
+- All submissions will be manually reviewed by the L3AF team
+- Once approved, programs will be published in the eBPF Package Repository
+
+Another important thing to note is that, initially, code submissions will need
+to conform to L3AF's eBPF program chaining mechanics.
+
+## To Build or Not to Build
+
+Because of the overhead and support requirements of a full build system, it may not be
+feasible to build the eBPF program source code in the initial version. However, we believe
+this would greatly hinder adoption from both contributors and users. Our proposal, therefore,
```

Comment:
what is "this" referring to?   having to build from source?  The word "However" seems confusing if so.

Reply:
@dthaler "this" means "not having to build from source".

Reply:
```suggestion
not being able to build from source would greatly hinder adoption from both contributors and users. Our proposal, therefore,
```